ml-prims/src/hmm/magma/b_likelihood.h

```diff
@@ -0,0 +1,242 @@
+#pragma once
+
+#include <stats/sum.h>
+
+#include "hmm/magma/b_bilinear.h"
+#include "hmm/magma/b_inverse.h"
+#include "hmm/magma/b_determinant.h"
+
+using namespace MLCommon;
+using namespace MLCommon::LinAlg;
+
+namespace MLCommon {
+template <typename math_t>
+void log(math_t *out, const math_t *in, int len,
+         cudaStream_t stream = 0) {
+        unaryOp(out, in, len,
+                [] __device__(math_t in) {
+                        return std::log(in);
+                },
+                stream);
+}
+
+
+
+template <typename T>
+__global__
+void createLlhdBatchesKernel(int nObs, int nCl,
+                             T **dX_batches, T **dmu_batches, T **dInvSigma_batches,
+                             T **dX_array, T **dmu_array, T **dInvSigma_array,
+                             int nThreads_x, int nThreads_y){
+        int i_start = threadIdx.x + blockDim.x * blockIdx.x;
+        int j_start = threadIdx.y + blockDim.y * blockIdx.y;
+
+        for (size_t obsId = j_start; obsId < nObs; obsId+=nThreads_y) {
+                for (size_t clId = i_start; clId < nCl; clId+=nThreads_x) {
+
+                        size_t idx = IDX(clId, obsId, nCl);
+                        dX_batches[idx] = dX_array[obsId];
+                        dmu_batches[idx] = dmu_array[clId];
+                        dInvSigma_batches[idx] = dInvSigma_array[clId];
+                }
+        }
+}
+
+template <typename T>
+void create_llhd_batches(int nObs, int nCl,
+                         T **&dX_batches, T **&dmu_batches, T **&dInvSigma_batches,
+                         T **&dX_array, T **&dmu_array, T **&dInvSigma_array){
```

Comment:
accept stream argument

Reply:
I have moved this file for the next GMM PR